### DISCUSSION TITLE
Github API list of users, their languages and their followers

### DISCUSSION THREAD

**Discussion by sahraxyz** [community/community#21896] [New to GitHub]
<p>Hi, </p>
<p>I’m collecting data on github users, the languages they use and their collaboration, i.e. who they follow, followers and people they work with. </p>
<p>I am using a query similar to the one below:</p>
<pre><code>https://api.github.com/search/users?q=repos:%3E12+followers:%3C1000&amp;location:uk+language:python&amp;page=1&amp;per_page=100
</code></pre>
<p>Output:</p>
<pre><code>{
  "total_count": 1347162,
  "incomplete_results": false,
  "items": [
    {
      "login": "cyfdecyf",
      "id": 344294,
      "avatar_url": "https://avatars3.githubusercontent.com/u/344294?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/cyfdecyf",
      "html_url": "https://github.com/cyfdecyf",
      "followers_url": "https://api.github.com/users/cyfdecyf/followers",
      "following_url": "https://api.github.com/users/cyfdecyf/following{/other_user}",
      "gists_url": "https://api.github.com/users/cyfdecyf/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/cyfdecyf/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/cyfdecyf/subscriptions",
      "organizations_url": "https://api.github.com/users/cyfdecyf/orgs",
      "repos_url": "https://api.github.com/users/cyfdecyf/repos",
      "events_url": "https://api.github.com/users/cyfdecyf/events{/privacy}",
      "received_events_url": "https://api.github.com/users/cyfdecyf/received_events",
      "type": "User",
      "site_admin": false,
      "score": 1.0
    },
    {
      "login": "robertdavidgraham",
      "id": 3814757,
      "avatar_url": "https://avatars2.githubusercontent.com/u/3814757?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/robertdavidgraham",
      "html_url": "https://github.com/robertdavidgraham",
      "followers_url": "https://api.github.com/users/robertdavidgraham/followers",
      "following_url": "https://api.github.com/users/robertdavidgraham/following{/other_user}",
      "gists_url": "https://api.github.com/users/robertdavidgraham/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/robertdavidgraham/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/robertdavidgraham/subscriptions",
      "organizations_url": "https://api.github.com/users/robertdavidgraham/orgs",
      "repos_url": "https://api.github.com/users/robertdavidgraham/repos",
      "events_url": "https://api.github.com/users/robertdavidgraham/events{/privacy}",
      "received_events_url": "https://api.github.com/users/robertdavidgraham/received_events",
      "type": "User",
      "site_admin": false,
      "score": 1.0
    },
</code></pre>
<p>This isn’t giving me what I need except the user’s name. As you can see the user’s followers and follows are also not detailed. </p>
<p>I’ve been reading the documentation and looking around but to no avail - eventually I would like to turn this into a data set that has - user name, location, languages used, number of follows, number of followers.</p>
<p>Hope you can help out</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There isn’t an easy way to extract the information you want. Also, wholesale scraping of the GitHub userbase is generally considered bad form and may be against the <a href="https://help.github.com/articles/github-terms-of-service/#h-api-terms" rel="nofollow noopener">GitHub Terms of Service</a>.</p>
<p>If you’re doing academic research, we generally recommend you use <a href="https://blog.github.com/2017-01-19-github-data-ready-for-you-to-explore-with-bigquery/" rel="nofollow noopener">one of the datasets that GitHub has made public</a>. These are made available on platforms where you can query them to your heart’s content and they don’t put a load on GitHub’s servers, potentially impacting other customers.</p>
<p>I hope that helps!</p>

**Comment by sahraxyz** [comment]
<p>Thank you for this information - didn’t have any knowledge of it. </p>
